### PR TITLE
select: add MISSING operator support

### LINKED ISSUE #14358: Using "is not missing" clause in SelectObjectContent results in "invalid arithmetic operator"
## Expected Behavior
As per https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-glacier-select-sql-reference-select.html#s3-glacier-select-sql-reference-from, the "is not missing" clause can be used to check for omitted fields.

I uploaded a file with the following content to Amazon S3:
```
{"request":{"uri":"/1","header":{"User-Agent":"test"}}}
{"request":{"uri":"/2","header":{}}}
```

Querying the aforementioned file with the query `select * from s3object[*] as s where s.request.header['User-Agent'] is not missing` results in the first line being returned, as shown below:

```
$ aws s3api select-object-content --bucket $BUCKET --key test.json --input-serialization 'JSON={Type=LINES},CompressionType=NONE' --output-serialization 'JSON={}' --expression-type SQL --expression "select * from s3object[*] as s where s.request.header['User-Agent'] is not missing" output
$ cat output 
{"request":{"uri":"/1","header":{"User-Agent":"test"}}}
```

## Current Behavior
When using minio, running the aforementioned query results in an error:

```
$ aws s3api select-object-content --profile minio --endpoint-url $MINIO_ENDPOINT --bucket $BUCKET --key test.json --input-serialization 'JSON={Type=LINES},CompressionType=NONE' --output-serialization 'JSON={}' --expression-type SQL --expression "select * from s3object[*] as s where s.request.header['User-Agent'] is not missing" output

An error occurred (InternalError) when calling the SelectObjectContent operation: invalid arithmetic operator
```

## Possible Solution
Go isn't exactly my forte but referring to https://github.com/minio/minio/blob/master/internal/s3select/sql/parser.go I didn't find "missing" as being one of the recognized keywords, therefore I suspect it is not implemented at this time.

## Steps to Reproduce (for bugs)
Please see above.

## Context
I'm trying to query JSON files where a certain field is missing.

## Regression
No

## Your Environment
* Version used (`minio --version`): RELEASE.2022-02-18T01-50-10Z
* Server setup and configuration: Minio running in single filesystem mode
* Operating System and version (`uname -a`): ```Linux storage1 5.4.0-96-generic #109-Ubuntu SMP Wed Jan 12 16:49:16 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux```

### PR DESCRIPTION
**Pull request by klauspost**
## Description
Probably not full support, but for regular checks it should work.

Fixes https://github.com/minio/minio/issues/14358
Includes https://github.com/minio/minio/pull/14405

## Types of changes
- [x] New feature (non-breaking change which adds functionality)
- [x] Unit tests added/updated

### CODE REVIEW

File: internal/s3select/sql/value.go [hunk 1]
```diff
@@ -387,16 +405,25 @@ func (v *Value) compareOp(op string, a *Value) (res bool, err error) {
 	if !isValidComparisonOperator(op) {
 		return false, errArithInvalidOperator
 	}
+	fmt.Println("a:", a.GetTypeString(), op, "v:", v.GetTypeString())
```
Comment (harshavardhana) [inline internal/s3select/sql/value.go]: Looks like needs to be removed.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Comment by harshavardhana** [comment]
Looks like some test failures with simdj

```
ok  	github.com/minio/minio/internal/s3select/simdj	0.004s
sql.Missing{} (sql.Missing) != <nil> (<nil>)
--- FAIL: TestJsonpathEval (0.00s)
    jsonpath_test.go:93: case: 4 2 failed
FAIL
```

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by harshavardhana — APPROVED** [review]

**Review by donatello — APPROVED** [review]

**Pull request merged**